apps/omar-geoscript-app/grails-app/services/omar/geoscript/HeatMapService.groovy

```diff
@@ -90,10 +103,53 @@ class HeatMapService {
             {
                 if((isValidJson(result.hits.hits.getAt(i)._source.message))) {
                     Feature feature = writer.newFeature
-                    Map<String, Object> map = new ObjectMapper().readValue(result.hits.hits.getAt(i)._source.message, HashMap.class);
+                    Map<String, Object> logmap = new ObjectMapper().readValue(result.hits.hits.getAt(i)._source.message, HashMap.class);
+
+                    def minx = logmap.get("bbox").minX
+                    def miny = logmap.get("bbox").minY
+                    def maxx = logmap.get("bbox").minX
+                    def maxy = logmap.get("bbox").minY
+                    def srs = logmap.get("bbox").proj.id
+
+
+                    def width = logmap.get("params").width
+                    def height = logmap.get("params").height
+
+
+                    Bounds bounds = new Bounds(minx, miny, maxx, maxy)
+                    bounds.proj = srs
+
+
+                    def proc = new GeoScriptProcess( "vec:Heatmap" )
+
+                    def raster = proc.execute(
+                            data: writer,
+                            radiusPixels: 20,
+                            pixelsPerCell: 1,
+                            outputBBOX: bounds.env,
+                            outputWidth: width,
+                            outputHeight: height
+                    )?.result
+
+                    raster.sytle = new ColorMap( [
+                            [color: "#FFFFFF", quantity: 0, label: "nodata", opacity: 0],
```

Comment:
Let's pull these color string defs out into variables defined above.

Reply:
So you want "#FFFFFF", "#4444FF", etc to be variables?

Reply:
Made the color string defs variables.  (can be changed if necesary)